File: frontend/src/config/proveedor.js
/**
 * Quien presta el servicio, en un solo sitio.
 *
 * Estos datos NO son decorativos: la Ley 29571 obliga a que el Libro de
 * Reclamaciones identifique al proveedor -razon social, RUC y domicilio-, y
 * los terminos y la politica de privacidad tienen que decir con quien
 * contrata el cliente y a quien reclama. Antes vivian repetidos en tres
 * paginas como texto suelto, que es la forma segura de que un dia digan
 * cosas distintas.
 *
 * CargoXprez es la MARCA del producto; SOLUCIONES INFORMATICAS MDD S.A.C. es la
 * empresa que responde por el. Star Insights IT y SISAC son marcas de la misma
 * casa y se quedan donde son un credito de autoria, no una identificacion
 * legal.
 */
export const PROVEEDOR = {
  razonSocial: 'SOLUCIONES INFORMÁTICAS MDD S.A.C.',
  ruc: '20490042068',
  // Domicilio fiscal segun la ficha RUC. Va completo a proposito: "Puerto
  // Maldonado" a secas no identifica un domicilio para una reclamacion.
  domicilio: 'Av. Madre de Dios N° 1087, Dpto. 201, A.H. Huerto Familiar, Tambopata, Madre de Dios, Perú',
  ciudad: 'Puerto Maldonado, Madre de Dios',
  telefono: '(082) 573844',
  email: 'soporte@sisac.pe',
  web: 'https://sisac.pe',
  // El producto. Se separa de la razon social porque el cliente conoce esto,
  // no aquello.
  producto: 'CargoXprez',
};

/**
 * La marca, con los datos de su registro.
 *
 * Salen del Certificado N° 00165238 de INDECOPI (Resolucion 012389-2025/DSD,
 * 15 de abril de 2025). Se citan tal cual y no de memoria: un aviso de marca
 * que exagera lo que protege es peor que no ponerlo, porque es lo primero que
 * desmonta quien la use.
 *
 * DOS COSAS QUE CONVIENE TENER DELANTE, y por eso estan aqui y no solo en el
 * certificado guardado en un cajon:
 *
 *   - La titularidad es de TRES PERSONAS NATURALES, no de la empresa. Por eso
 *     el aviso NO dice "marca de SOLUCIONES INFORMATICAS MDD S.A.C.": diria
 *     algo que el registro no respalda. Que la titular sea la empresa es una
 *     cesion inscrita ante INDECOPI, no una frase en una web.
 *   - La clase 39 cubre transporte, embalaje, almacenamiento y organizacion de
 *     viajes. NO cubre software, que seria la clase 42. Usar la marca para
 *     nombrar este producto es legitimo, pero no esta protegida en esa clase.
 */
export const MARCA = {
  nombre: 'CargoXprez',
  certificado: 'N° 00165238',
  clase: 39,
  vigenciaHasta: '15 de abril de 2035',
};

/** El aviso de marca, para el pie y los textos legales. */
export const AVISO_DE_MARCA =
  `${MARCA.nombre}® es marca registrada ante INDECOPI (Certificado ` +
  `${MARCA.certificado}, clase ${MARCA.clase}), vigente hasta el ` +
  `${MARCA.vigenciaHasta}. Todos los derechos reservados.`;

/** "CargoXprez, servicio de SOLUCIONES INFORMÁTICAS MDD S.A.C. (RUC 20490042068)" */
export const proveedorEnUnaLinea = () =>
  `${PROVEEDOR.producto}, servicio de ${PROVEEDOR.razonSocial} (RUC ${PROVEEDOR.ruc})`;

File: backend/proveedor.py
"""Quien presta el servicio, en Python.

ESTO ES UN ESPEJO DE frontend/src/config/proveedor.js, NO UNA SEGUNDA VERDAD

  El checkout de React lee el fichero JS; el checkout del servidor
  (`checkout_sin_js.py`) no puede, porque vive en otro lenguaje y se renderiza
  antes de que exista un navegador. Los mismos datos hacen falta en los dos.

  Sin un paso de build que los comparta, la unica forma honesta de tener dos
  copias es OBLIGARLAS A COINCIDIR: `tests/test_proveedor_espejo.py` lee el JS
  y comprueba valor por valor. Si alguien corrige el telefono en uno solo, la
  prueba se cae, en vez de dejar dos paginas diciendo cosas distintas sobre a
  quien reclamar.

  Estos datos no son decorativos: la Ley 29571 obliga a identificar al
  proveedor -- razon social, RUC y domicilio -- y son los que mira quien valida
  el comercio antes de habilitar la pasarela.
"""

RAZON_SOCIAL = "SOLUCIONES INFORMÁTICAS MDD S.A.C."
RUC = "20490042068"
DOMICILIO = ("Av. Madre de Dios N° 1087, Dpto. 201, A.H. Huerto Familiar, "
             "Tambopata, Madre de Dios, Perú")
TELEFONO = "(082) 573844"
EMAIL = "soporte@sisac.pe"
PRODUCTO = "CargoXprez"

# El aviso de marca, con los datos del Certificado N° 00165238 de INDECOPI.
#
# OJO CON LA REDACCION, y la advertencia viene del propio proveedor.js: la
# titularidad es de TRES PERSONAS NATURALES, no de la empresa. Por eso el aviso
# dice "marca registrada ante INDECOPI" y NUNCA "marca de SOLUCIONES
# INFORMATICAS MDD S.A.C.", que seria afirmar algo que el registro no respalda.
MARCA_CERTIFICADO = "N° 00165238"
MARCA_CLASE = 39


def en_una_linea() -> str:
    """El mismo texto que compone `proveedorEnUnaLinea()` en el JS."""
    return f"{PRODUCTO}, servicio de {RAZON_SOCIAL} (RUC {RUC})"


def telefono_uri() -> str:
    """El numero en forma internacional, para que un movil pueda marcarlo.

    El "0" de "(082)" es el prefijo interurbano peruano y solo vale marcando
    dentro del pais: pegado a +51 daria +51082..., que no existe. Se descarta.
    """
    digitos = "".join(c for c in TELEFONO if c.isdigit())
    return "+51" + digitos.lstrip("0")
